telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java

```diff
@@ -118,19 +124,11 @@ private void retry(
   }
 
   private void backoff(
-      BatchSender sender,
-      TelemetryBatch<? extends Telemetry> batch,
-      int preWaitTime,
-      TimeUnit timeUnit) {
-    int newWaitTime;
-    if (preWaitTime == 0) {
-      newWaitTime = 1;
-      timeUnit = TimeUnit.SECONDS;
-    } else {
-      newWaitTime = preWaitTime * 2;
-    }
-    LOG.info("Metric batch sending failed. Backing off {} {}", newWaitTime, timeUnit);
-    scheduleBatchSend(sender, batch, newWaitTime, timeUnit);
+      BatchSender sender, TelemetryBatch<? extends Telemetry> batch, Backoff backoff) {
+
+    long newWaitTime = backoff.nextWaitMs();
+    LOG.info("Metric batch sending failed. Backing off {} {}", newWaitTime, TimeUnit.MILLISECONDS);
```

Comment:
I think you're missing the handling of the -1 case here.

Reply:
I don't follow what you mean by this.

Reply:
Oh NM, I see what you mean now and I think you're right.